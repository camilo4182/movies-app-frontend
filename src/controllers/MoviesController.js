import axios from 'axios';

export const MoviesController = {
    register(movie) {
        return axios({
            method: 'POST',
            baseURL: process.env.API,
            url: 'movies',
            data: movie
        });
    },
    update(movie_title){
        return axios({
            method: 'PUT',
            baseURL: process.env.API,
            url: `movies/${movie_title}`
        });
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'movies'
        });
    },
    delete(movie_title) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `movies/${movie_title}`
        });
    }
}

