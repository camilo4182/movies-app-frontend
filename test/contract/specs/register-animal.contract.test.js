import {provider} from "../specs/init-pact";
import {AnimalController} from "../../../controllers";
import {Matchers} from "@pact-foundation/pact";

describe('Given an animal service', () => {
    describe('When a request to register an animal is made', () => {
        const animal = {
            name: "Pelusa",
            breed: "Siames",
            gender: "Male",
            isVaccinated: true
        }
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'register an animal',
                uponReceiving: 'a request to register an animal',
                withRequest: {
                    method: 'POST',
                    path: '/animals',
                    body: animal
                },
                willRespondWith: {
                    status: 201,
                    headers:{
                        'Content-Type': 'application/json'
                    },
                    body: Matchers.somethingLike(
                        {
                            name: Matchers.like('manchas'),
                            breed: Matchers.like("Bengali"),
                            gender: Matchers.like("Female"),
                            vaccinated: Matchers.boolean(true)
                        }
                    )
                }
            });
        });

        it('Then it should return the right http code', async () => {
            const response = await AnimalController.register(animal);
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

        afterAll(async () => {
            await provider.finalize();
        })
    });
});