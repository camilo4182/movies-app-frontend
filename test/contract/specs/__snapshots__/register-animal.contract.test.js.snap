// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an animal service When a request to register an animal is made Then it should return the right http code 1`] = `
Object {
  "breed": "Bengali",
  "gender": "Female",
  "name": "manchas",
  "vaccinated": true,
}
`;
